src/python/grapl_analyzerlib/grapl_analyzerlib/nodes/process_user.py

```diff
@@ -0,0 +1,94 @@
+from collections import defaultdict
+from typing import Any, TypeVar, List, Set, Type, Dict, Tuple, Optional, Iterator, Union
+
+from grapl_analyzerlib.node_types import (
+    EdgeT,
+    PropType,
+    PropPrimitive,
+    EdgeRelationship,
+)
+from grapl_analyzerlib.queryable import Queryable, EdgeFilter, ToOneFilter, ToManyFilter
+from grapl_analyzerlib.schema import Schema
+from grapl_analyzerlib.viewable import Viewable, V, Q
+from grapl_analyzerlib.comparators import StrCmp, Eq, Distance
+
+from grapl_analyzerlib.nodes.process import (
+    ProcessQuery,
+    ProcessView,
+    ProcessSchema,
+    PQ,
+    PV,
+)
+
+
+class ProcessWithUserQuery(ProcessQuery):
+    def __init__(self):
+        super(ProcessWithUserQuery, self).__init__()
+        self._user: List[List[StrCmp]] = []
+        ProcessQuery.viewable = ProcessQuery.viewable.extend_self(ProcessWithUserView)
+
+    def with_user(
+        self: PQ,
+        eq: Optional[str] = None,
+        contains: Optional[Union[str, List[str]]] = None,
+        starts_with: Optional[str] = None,
+        ends_with: Optional[str] = None,
+        regexp: Optional[Union[str, List[str]]] = None,
+        distance_lt: Optional[Tuple[str, int]] = None,
+    ) -> PQ:
+        self._property_filters["user"].extend(
+            _str_cmps(
+                predicate="user",
+                eq=eq,
+                contains=contains,
+                ends_with=ends_with,
+                starts_with=starts_with,
+                regexp=regexp,
+                distance_lt=distance_lt,
+            )
+        )
+        return self
+
+
+class ProcessWithUserView(ProcessView):
+    user = None
```

Comment:
(is this needed? i find mutable class variables to be more trouble than they're worth)

Reply:
It helps pytype infer some things. I'll see if I can remove it. I'm with you though.